i18n/check_and_integrate_translations.py

```diff
@@ -0,0 +1,154 @@
+    print("        -S, --skip-sync : Skip the synchronization and merge of PO files with the reference POT files.")
+    sys.exit()
+
+
+def main(argv):
+    # Recovering options and arguments from command line
+    opts = list()
+    try:
+        opts, args = getopt(argv, "ht:i:ES", ["help", "translations_path=", "integrate=", "skip-extract", "skip-sync"])
+    except GetoptError:
+        help_output()
+    # Converting list of tuples in a convenient dictionary
+    opts_dict = {opt[0]: opt[1] for opt in opts}
+
+    # Getting Translations Directory Path from options
+    translations_path = ''
+    if '-t' in opts_dict.keys():
+        translations_path = opts_dict['-t']
+    elif '--trans_path' in opts_dict.keys():
+        translations_path = opts_dict['--trans_path']
+
+    # Getting Threshold value from options
+    if '-i' in opts_dict.keys():
+        try:
+            threshold = int(opts_dict["-c"])
+        except ValueError:
+            print("WARN: Incorrect Threshold value. Script will proceed without languages integration.")
+            threshold = None
+    elif '--integrate' in opts_dict.keys():
+        try:
+            threshold = int(opts_dict["--completion_threshold"])
+        except ValueError:
+            print("WARN: Incorrect Threshold value. Script will proceed without languages integration.")
+            threshold = None
+    else:
+        print("WARN: Missing Threshold value. Script will proceed without languages integration.")
+        threshold = None
+
+    # Triggering Help message if translations path missing or help options present.
+    if translations_path == '' or not os.path.exists(translations_path)\
+            or len(opts) == 0 or '-h' in opts_dict.keys() or '--help' in opts_dict.keys():
+        help_output()
+
+    # Prevent execution from another directory than script source directory
+    source_dir = os.getcwd()
+    if not source_dir.endswith('learn-gdscript'):
+        sys.exit("ERROR: This script should be executed from learn-gdscript project directory.")
+
+    # Generating POT files from extract.py
+    extract_result = -1
+    if not ('-E' in opts_dict.keys() or '--skip-extract' in opts_dict.keys()):
+        print("INFO: Extracting strings and generating POT files...")
+        extract_result = system("python3 i18n/extract.py")
+
+        if extract_result:
+            sys.exit("ERROR: Extraction scripts ended with errors")
+
+        # Moving POT files to translations project
+        print("INFO: Moving POT files to translations folder")
+        pot_files = [file for file in os.listdir('i18n') if file.endswith(".pot")]
+        for pot_file in pot_files:
+            move(os.path.join('i18n', pot_file), os.path.join(translations_path, pot_file))
+    else:
+        print("WARN: Skipping strings extraction and POT files generation...")
+
+    # Updating PO files with sync_translations.py
+    os.chdir(translations_path)
```

Comment:
I recommend not changing the directory in those kinds of scripts for two reasons:

1. It's error-prone. For example, this script changes your directory to another working directory, but it doesn't change it back to the source place you were at the end.
2. Doesn't work if you want to integrate into other workflows or CI.

Reply:
Thanks for this advice, I'll keep it in mind next time I'll develop a system script !